compiler/transforms/convert_linalg_to_kernel.py

```diff
@@ -0,0 +1,73 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, linalg
+from xdsl.ir import Block
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint
+
+from compiler.dialects.kernel import Kernel, Parsable
+
+
+def check_block(block_a: Block, block_b: Block) -> bool:
+    """
+    Verify if two blocks are equal to each other,
+    that for the same inputs they include the same
+    operations.
+    """
+    if len(block_a.ops) != len(block_b.ops):
+        return False
+
+    # warning: this is a bit of a naive way of checking equality between
+    # kernels, but should cover all of our purposes for quite some time
+    for op_a, op_b in zip(block_a.ops, block_b.ops, strict=True):
+        if type(op_a) is not type(op_b):
+            return False
+
+    return True
+
+
+class ParseLinalgBody(RewritePattern):
+    """
+    Matches on linalg.generic operations to check if
+    their body matches a specific kernel op defined in
+    the kernel dialect. Replaces the body with the relevant
+    kernel op if this is true.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, linalg_op: linalg.Generic, rewriter: PatternRewriter):
+        for op_def in Kernel.operations:
+            if not issubclass(op_def, Parsable):
+                continue
```

Comment:
Is there any op we would define here that is not `Parsable`?

Reply:
yes, kernel.rescale from https://github.com/KULeuven-MICAS/snax-mlir/pull/236 for example